test/test_runner.go

```diff
@@ -144,7 +145,13 @@ type TestRunner struct {
 	// log messages from test cases and contracts.
 	logCollection *LogCollectionHook
 
-	backend *EmulatorBackend
+	// blockchain is mainly used to obtain system-defined
+	// contracts & their exposed types
+	blockchain *emulator.Blockchain
```

Comment:
should this field be of type `stdlib.Blockchain`?
```suggestion
	blockchain stdlib.Blockchain
```

Reply:
Just realized, we are creating a new blockchain for every call of `newEmulatorBlockchain`, correct? Can we get these required details (default contracts and their types) from that created blockchain instance, instead of maintaining this unused/empty blockchain just for that?

Reply:
Yes, exactly, with every call of `Test.newEmulatorBlockchain` we are also creating a new `EmulatorBackend`, which in turn creates a new `Blockchain`.

I do not know if that's possible, because when `interpreterImportHandler` is run, the `Test.newEmulatorBlockchain` has not yet been called/interpreted, and there is no `Blockchain` instance that we can use to get the default contracts and their types. Maybe I am missing something though.

Reply:
The `EmulatorBackend` is no longer created by the `TestRunner`, as it used to be, so I am not sure how we can get a grip on it. The `TestRunner` only creates the `stdlib.TestFramework` implementation.

Reply:
Is it only for system-defined / default contracts? Can we somehow include that as part of the import resolver?

Reply:
i.e: Can we do parsing and type checking of all imports at 'parseAndCheckImport`? Currently, there are multiple paths where imports are resolved, and some paths have custom logics (E.G. `interpreterImportHandler`), while others don't `checkerImportHandler`, which could create inconsistencies. Might be good to unify them all.

Reply:
Great point Supun :100:  I have updated this in https://github.com/onflow/cadence-tools/pull/205/commits/aa3d7c89512ee9a1830d19de554b8afb94e12d12. It removed quite a lot of code.

I opted to keep the `*emulator.Blockchain`, because in my opinion, it will make the code base more resilient to additions of system contracts. I'm not sure if this can be implemented in `importResolver`, without listing every single system contract, which means that an update will be needed every time a system contract is added/removed.
What do you think?